### PR TITLE
docs: YARD method and class documentation.

### LINKED ISSUE #9: YARD method/class documentation
- [x] YARD documentation. params, exceptions, return values.
- [x] Make sure it renders correctly

### PR DESCRIPTION
**Pull request by schwern**
YARD is a Ruby documentation tool. The relevant docs are [Getting Started](https://rubydoc.info/gems/yard/file/docs/GettingStarted.md) and [Tag Overview](https://rubydoc.info/gems/yard/file/docs/Tags.md). This completes the YARD documentation of methods and classes. It is not complete documentation for the library, [there's plenty more](https://github.com/momentohq/client-sdk-ruby/milestone/1).

If you don't know Ruby or YARD, please check over the generated docs: [doc.zip](https://github.com/momentohq/client-sdk-ruby/files/10135604/doc.zip). Download and unzip it, and open `doc/index.html` in a browser.

If you want to generate the docs...

* Run `bundle` to update your development dependencies and install yard.
* Run `yard doc` to generate the docs.
* Open `doc/index.html` in a browser.

What this does...

* Configure YARD.
  * Skip private methods and classes
  * Skip the protocol buffer files
* Update the README so it renders well in YARD.
* Hide internal classes, modules, and constants
* Document every visible method and class
  * What it returns; what it raises; params, attributes, and their types
* `@!method method_name` is used to document an inherited method in a subclass.
* Expand the SimpleCacheClient example.
* Add all the subclass methods to ListCachesResponse for discoverability.

Closes #9

### CODE REVIEW

File: lib/momento/get_response.rb [hunk 1]
```diff
@@ -1,14 +1,19 @@
 require_relative 'response/error'
 
 module Momento
-  # Responses specific to get.
+  # A response from getting a key.
+  #
+  # Because Momento only stores bytes, the value will be
+  # encoded using Encoding::ASCII_8BIT.
```
Comment (malandis) [inline lib/momento/get_response.rb:7]: ASCII as opposed to UTF8? The latter is the standard in other SDKs. Is there a Ruby specific issue here?

**Review by malandis — COMMENTED** [review]
I reviewed the documentation and it looks good; no typos AFAICT. One question about a string encoding...

File: README.md [hunk 2]
```diff
@@ -73,8 +77,8 @@ else
 end
 ```
 
-You can run this code from [examples/basic.rb](examples/basic.rb).
+You can find this code in [the examples directory](https://github.com/momentohq/client-sdk-ruby/tree/main/examples).
 
 ## Contributing
 
-See our [instructions for CONTRIBUTING](./CONTRIBUTING.md).
+See our [instructions for CONTRIBUTING](https://github.com/momentohq/client-sdk-ruby/blob/main/CONTRIBUTING.md).
```
Comment (cprice404) [inline README.md:84]: just curious why absolute here?

**Review by cprice404 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (schwern) [reply README.md:84]: When YARD generates the HTML it does not know how to turn it into an absolute URL.

Comment (cprice404) [reply README.md:84]: so YARD is going to generate stuff from the README as well as the ruby code?  If that's the case, TIL, 🚢

File: lib/momento/simple_cache_client.rb [hunk 3]
```diff
@@ -9,25 +9,44 @@ module Momento
   # A simple client for Momento.
   #
   # @example
```
[the remark this answers is not in the document]
Comment (schwern) [reply lib/momento/simple_cache_client.rb:11]: Not that I know of. It wouldn't be hard to write our own thing to extract the examples and test them. But I'll [ask on StackOverflow](https://stackoverflow.com/questions/74648399/can-i-get-all-examples-from-my-yard-docs).

File: README.md (hunk 2 above)
Comment (schwern) [reply README.md:84]: Yes. For example, YARD's own README.

https://github.com/lsegal/yard/blob/main/README.md
https://rubydoc.info/gems/yard/file/README.md

**Pull request merged**